src/main/scala/com/johnsnowlabs/nlp/util/io/ResourceHelper.scala

```diff
@@ -56,33 +56,45 @@ object ResourceHelper {
   def getSparkSessionWithS3(
       awsAccessKeyId: String,
       awsSecretAccessKey: String,
-      awsSessionToken: String): SparkSession = {
+      hadoopAwsVersion: String,
+      region: String = "us-east-1",
+      s3Impl: String = "org.apache.hadoop.fs.s3a.S3AFileSystem",
+      pathStyleAccess: Boolean = true,
+      credentialsProvider: String = "TemporaryAWSCredentialsProvider",
+      awsSessionToken: Option[String] = None): SparkSession = {
 
     require(
       SparkSession.getActiveSession.isEmpty,
       "Spark session already running, can't apply new configuration for S3.")
 
-    SparkSession
+    val sparkSession = SparkSession
       .builder()
       .appName("SparkNLP Session with S3 Support")
       .master("local[*]")
       .config("spark.driver.memory", "22G")
       .config("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
-      .config("spark.kryoserializer.buffer.max", "2000M")
+      .config("spark.kryoserializer.buffer.max", "1000M")
       .config("spark.driver.maxResultSize", "0")
       .config("spark.jsl.settings.aws.credentials.access_key_id", awsAccessKeyId)
       .config("spark.jsl.settings.aws.credentials.secret_access_key", awsSecretAccessKey)
-      .config("spark.jsl.settings.aws.credentials.session_token", awsSessionToken)
-      .config("spark.jsl.settings.aws.region", "us-east-1")
+      .config("spark.jsl.settings.aws.region", region)
       .config(
         "spark.hadoop.fs.s3a.aws.credentials.provider",
-        "org.apache.hadoop.fs.s3a.TemporaryAWSCredentialsProvider")
-      .config("spark.hadoop.fs.s3a.impl", "org.apache.hadoop.fs.s3a.S3AFileSystem")
+        s"org.apache.hadoop.fs.s3a.$credentialsProvider")
+      .config("spark.hadoop.fs.s3a.impl", s3Impl)
       .config(
         "spark.jars.packages",
-        "org.apache.hadoop:hadoop-aws:3.3.1,com.amazonaws:aws-java-sdk:1.11.901")
-      .config("spark.hadoop.fs.s3a.path.style.access", "true")
-      .getOrCreate()
+        "org.apache.hadoop:hadoop-aws:" + hadoopAwsVersion + ",com.amazonaws:aws-java-sdk:1.11.901")
```

Comment:
Good job defining arguments with a default value. We can add also an argument for `aws-java-sdk` version (like you did for hadoopAwsVersion)